backend/src/services/auto_trader.py

```
@@ -33,238 +23,431 @@ def __init__(self, socketio=None, data_fetcher_service=None, ai_analysis_service
             if self.socketio:
                 self.socketio.emit('trading_status', {'enabled': True, 'message': 'Automated trading started.'})
 
     def stop_trading(self):
         if self.trading_enabled:
             self.trading_enabled = False
             if self.trade_loop_task:
-                self.trade_loop_task.cancel()
+                try:
+                    self.trade_loop_task.cancel()
+                    logger.info("AutoTrader: Cancelled trade loop task.")
+                except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)